scripts/generate_activity_script/src/script.ts

```diff
@@ -0,0 +1,353 @@
+   */
+  let owned_nfts = (await withdrawal_voucher.queryTokens({ owner: address }))
+    .tokens;
+  let current_nft_list = owned_nfts;
+  while (current_nft_list.length !== 0) {
+    current_nft_list = (
+      await withdrawal_voucher.queryTokens({
+        owner: address,
+        start_after: current_nft_list[current_nft_list.length - 1],
+      })
+    ).tokens;
+    owned_nfts = owned_nfts.concat(current_nft_list);
+  }
+  const owned_nfts_detailed = [];
+  for (const nft of owned_nfts) {
+    owned_nfts_detailed.push({
+      nft_id: nft,
+      details: {
+        ...(await withdrawal_voucher.queryNftInfo({
+          token_id: nft,
+        })),
+      },
+    });
+  }
+  const withdrawn_nfts = [];
+  for (const nft_detailed of owned_nfts_detailed) {
+    if (
+      (
+        await drop_instance.queryUnbondBatch({
+          batch_id: nft_detailed.details.extension.batch_id,
+        })
+      ).status === "withdrawn"
+    ) {
+      withdrawn_nfts.push(nft_detailed.nft_id);
+    }
+  }
+
+  /* Nothing to withdraw
+   * Return null and try another method in contract
+   */
+  if (withdrawn_nfts.length === 0) {
+    return null;
+  }
+
+  /* Pick up random NFT from given NFT list and try to withdraw it
+   * If any error occured when executing method then just ignore
+   * It's content and return null, script will try to call another method
+   */
+  const nft_id: string =
+    withdrawn_nfts[Math.floor(Math.random() * withdrawn_nfts.length)];
+  try {
+    return await send_nft(
+      withdrawal_voucher,
+      withdrawal_manager,
+      address,
+      nft_id
+    );
+  } catch (e) {
+    return null;
+  }
+}
+
+async function main() {
+  const neutronWallet: Wallet = {};
+  neutronWallet.mainWallet = await DirectSecp256k1HdWallet.fromMnemonic(
+    MNEMONIC,
+    {
+      prefix: "neutron",
+    }
+  );
+  neutronWallet.clientCW = await SigningCosmWasmClient.connectWithSigner(
+    NEUTRON_NODE_ADDRESS,
+    neutronWallet.mainWallet,
+    {
+      gasPrice: GasPrice.fromString("0.75untrn"),
+    }
+  );
+  neutronWallet.clientSG = await SigningStargateClient.connectWithSigner(
+    NEUTRON_NODE_ADDRESS,
+    neutronWallet.mainWallet,
+    {
+      gasPrice: GasPrice.fromString("0.75untrn"),
+    }
+  );
+  neutronWallet.mainAccounts = await neutronWallet.mainWallet.getAccounts();
+
+  let core_contract = new DropCoreClient(neutronWallet.clientCW, CORE_CONTRACT);
+  const logs: Array<Action> = [];
+  if (Math.random() <= WITHDRAW_PROB) {
+    const res = await withdraw_random_nft(neutronWallet, core_contract);
+    if (res !== null) {
+      logs.push(res);
+    }
+  }
+  if (Math.random() <= UNBOND_PROB) {
+    const res = await unbond_random_amount(neutronWallet, core_contract);
+    if (res !== null) {
+      logs.push(res);
+    }
+  }
+  if (Math.random() <= BOND_PROB) {
+    const res = await bond_random_amount(neutronWallet, core_contract);
+    if (res !== null) {
+      logs.push(res);
+    }
+  }
+  console.log(logs);
```

Comment:
I don't get why you collect logs and log them once in the end instead of logging each step

Reply:
Because it's an array of logs. We need it in valid JSON format